### ISSUE TITLE
dietpi-upgrade 6.25.3 -> 6.32.2: dietpi-ramdisk: No such file or directory

### ISSUE THREAD

**Issue by badgertastic** [MichaIng/DietPi#3812]
#### Details:
- Date           | Tue  6 Oct 15:13:24 BST 2020
- Bug report     | N/A
- DietPi version | v6.25.3 (Fourdee/master)
- Img creator    | DietPi Core Team
- Pre-image      | Raspbian Lite
- SBC device     | RPi 3 Model B (armv7l) (index=3)
- Kernel version | #1253 SMP Thu Aug 15 11:49:46 BST 2019
- Distro         | stretch (index=4)
- Command        | /DietPi/dietpi/func/dietpi-ramdisk 1
- Exit code      | 127
- Software title | DietPi-Update

#### Steps to reproduce:

1. dietpi-update
(6.25.3 -> 6.32.2)

#### Expected behaviour:

- no error/failure

#### Actual behaviour:

- error: /DietPi/dietpi/func/dietpi-globals: line 1316: /DietPi/dietpi/func/dietpi-ramdisk: No such file or directory

#### Extra details:

- same on 2 separate Pi updates.

#### Additional logs:
```
Log file contents:
/DietPi/dietpi/func/dietpi-globals: line 1316: /DietPi/dietpi/func/dietpi-ramdisk: No such file or directory
```

**Comment by badgertastic** [comment]
System is running ok, reboots ok etc. just concern that Ram is no-longer used and SD card being used instead (eventually leading to failure/corruption/performance degrade of SD). Please comment if dietpi-ramdisk is now obsolete and what replaced it if so. thanks for all your support :)

**Comment by Joulinar** [comment]
Hi,

many thanks for your report. If I'm not mistaken this can be ignored. Isn't it @MichaIng 

maybe similar to this?? https://github.com/MichaIng/DietPi/issues/3532#issuecomment-628528041
Because scripts moved from `/DietPi` > `/boot` ??

**Comment by MichaIng** [comment]
Many thanks for your report.

First of all: Yes `DietPi-RAMdisk` has been removed. This included only the `/boot/dietpi.txt`, `/boot/dietpi` and `/boot/config.txt` on RPi. The SD writes are likely lower now since in the past, the whole RAMdisk content was written to disk on every reboot and after every update (as it was attempted on your case), now only when one actually changes a setting, the single config file is written. But it depends a bid und usage of course. The main purpose of the RAMdisk was not to reduce SD writes (those are minimal for config files) but to speed up SD reads, i.e. DietPi scripts and settings, and this is now assured via `vmtouch` service, which keeps those files locked in native file system (read-)cache. The reason for this was to simplify our boot and shutdown process (no need to copy back and forth the RAMdisk content) and to eliminate the risk that a system crash or unclean shutdown leads to lost config changes and in case broken/reverted updates. So it should be all more solid now.

And yes, we had this update issue one or two times before, which occurred because the updater somehow did not kill itself (more precisely, its parent script) as intended and <s>I am still now sure how this can happen</s> **EDIT: it has been fixed meanwhile, lol forgot that we found the reason and fixed it already 😄**. But as Joulinar said, it can be ignored since the updater already finished. It is an obsolete parent process which should have been killed but wasn't and now finished with outdated post-update steps, e.g. writing the RAMdisk content to disk which is not required and obviously not possible anymore.

But just to be very sure, do you still have the updater log in terminal, or the related log file:
```
cat /var/tmp/dietpi/logs/dietpi-update.log
```
________
Ah yes all makes sense, the v6.25-only bug where I moved the output split for log file into a pipe, but the pipe input processes looses control over the parent shell, hence cannot kill it anymore. Learning never ends 😄.

**Comment by badgertastic** [comment]
Thanks for the timely and informative post!  The situation is exactly as you describe and the installer did complete successfully. I'll paste the log FYI in any case. thanks again.

```
[  OK  ] DietPi-Patch | Everything done! Terminating the obsolete DietPi-Update parent instance...


 DietPi-Update
─────────────────────────────────────────────────────
 Mode: Completed

[ INFO ] DietPi-Update | Current version : v6.25.3 (branch: Fourdee/master)
[ INFO ] DietPi-Update | Latest version  : v6.32.2 (branch: MichaIng/master)
shell-init: error retrieving current directory: getcwd: cannot access parent directories: No such file or directory
chdir: error retrieving current directory: getcwd: cannot access parent directories: No such file or directory
[  OK  ] DietPi-Survey | Purging survey data
[FAILED] DietPi-Update | /DietPi/dietpi/func/dietpi-ramdisk 1
[FAILED] DietPi-Update | /DietPi/dietpi/func/dietpi-ramdisk 1
[FAILED] DietPi-Update | /DietPi/dietpi/func/dietpi-ramdisk 1


--------------------------------------------------------------------
- DietPi has encounted an error, and, is unable to continue        -
- Please create a ticket: https://github.com/MichaIng/DietPi/issues -
- Copy and paste the BLUE lines below, into the ticket             -
--------------------------------------------------------------------
```

Details as previously shared.

The log:
<details> 
  <summary>Logfile attached. Click to expand!</summary>

```
# cat /var/tmp/dietpi/logs/dietpi-update.log
[ SUB2 ] DietPi-Update > Applying pre-patches
[  OK  ] DietPi-Update | Downloading pre-patch file
[  OK  ] DietPi-Update | Applying execute permission
[ INFO ] Pre-patch 13 | Disabling dpkg status file translations by default, which have hardly any use, but raise APT update times + disk I/O 
[ INFO ] Pre-patch 20 | Mark ifupdown as manually installed to prevent its autoremoval
ifupdown was already set to manually installed.
[  OK  ] DietPi-Update | Successfully applied pre-patches
[ SUB2 ] DietPi-Update > Upgrading APT packages
[ INFO ] DietPi-Update | APT update, please wait...
Hit:1 http://raspbian.raspberrypi.org/raspbian stretch InRelease
Hit:2 https://archive.raspberrypi.org/debian stretch InRelease
Reading package lists...
[  OK  ] DietPi-Update | APT update
[ INFO ] DietPi-Update | APT upgrade, please wait...
[  OK  ] DietPi-Update | APT upgrade
[ SUB2 ] DietPi-Update > Installing new DietPi code
[  OK  ] DietPi-Update | Downloading update archive
[  OK  ] DietPi-Update | Unpacking update archive
[  OK  ] DietPi-Update | Installing new DietPi scripts
[  OK  ] DietPi-Update | Installing new DietPi system files
[  OK  ] DietPi-Update | Setting execute permissions for DietPi scripts
[ SUB3 ] DietPi-Set_software > verify_dietpi.txt ()
[  OK  ] DietPi-Set_software | Checking URL: https://raw.githubusercontent.com/MichaIng/DietPi/master/dietpi.txt
[  OK  ] DietPi-Set_software | Downloading current dietpi.txt
[ INFO ] DietPi-Set_software | Updating dietpi.txt with new entry: AUTO_SETUP_ACCEPT_LICENSE=0
[ INFO ] DietPi-Set_software | Updating dietpi.txt with new entry: AUTO_SETUP_NET_WIFI_COUNTRY_CODE=GB
[ INFO ] DietPi-Set_software | Updating dietpi.txt with new entry: CONFIG_CHECK_CONNECTION_IP=9.9.9.9
[ INFO ] DietPi-Set_software | Updating dietpi.txt with new entry: CONFIG_CHECK_DNS_DOMAIN=dns9.quad9.net
[ INFO ] DietPi-Set_software | Updating dietpi.txt with new entry: SOFTWARE_HOMEASSISTANT_APT_DEPS=
[ INFO ] DietPi-Set_software | Updating dietpi.txt with new entry: SOFTWARE_HOMEASSISTANT_PIP_DEPS=
[  OK  ] DietPi-Set_software | dietpi.txt verification completed
[  OK  ] verify_dietpi.txt  | Completed
[ SUB2 ] DietPi-Update > Applying incremental patches
[ INFO ] DietPi-Update | Current version : v6.25.3
[ INFO ] DietPi-Update | Latest version  : v6.32.2
[ INFO ] DietPi-Patch | Patching 6.25 to 6.26
[ SUB4 ] DietPi-Set_hardware > rpi-camera (disable)
[  OK  ] rpi-camera disable | Completed
[  OK  ] DietPi-Patch | Added setting dtoverlay=disable-bt to end of file /boot/config.txt
[  OK  ] DietPi-Patch | Added setting dtoverlay=disable-wifi to end of file /boot/config.txt
[  OK  ] DietPi-Patch | Comment in /boot/config.txt converted to setting: initial_turbo=20
[ INFO ] DietPi-Patch | APT purge for: haveged, please wait...
(Reading database ... 28126 files and directories currently installed.)
Removing haveged (1.9.1-5+deb9u1) ...
(Reading database ... 28117 files and directories currently installed.)
Purging configuration files for haveged (1.9.1-5+deb9u1) ...
Processing triggers for systemd (232-25+deb9u12) ...
[  OK  ] DietPi-Patch | APT purge for: haveged
[ INFO ] DietPi-Patch | APT install for: rng-tools, please wait...
Selecting previously unselected package rng-tools.
(Reading database ... 28113 files and directories currently installed.)
Preparing to unpack .../rng-tools_2-unofficial-mt.14-1_armhf.deb ...
Unpacking rng-tools (2-unofficial-mt.14-1) ...
Setting up rng-tools (2-unofficial-mt.14-1) ...
Processing triggers for systemd (232-25+deb9u12) ...
[  OK  ] DietPi-Patch | APT install for: rng-tools
[ SUB4 ] DietPi-Set_hardware > headless (enable)
[  OK  ] DietPi-Set_hardware | Comment in /boot/config.txt converted to setting: framebuffer_width=16
[  OK  ] DietPi-Set_hardware | Comment in /boot/config.txt converted to setting: framebuffer_height=16
[  OK  ] DietPi-Set_hardware | Comment in /boot/config.txt converted to setting: max_framebuffer_width=16
[  OK  ] DietPi-Set_hardware | Comment in /boot/config.txt converted to setting: max_framebuffer_height=16
[  OK  ] DietPi-Set_hardware | Desired setting in /boot/config.txt was already set: disable_splash=1
[  OK  ] DietPi-Set_hardware | Desired setting in /boot/config.txt was already set: hdmi_blanking=1
[  OK  ] DietPi-Set_hardware | Desired setting in /boot/config.txt was already set: hdmi_ignore_hotplug=1
[  OK  ] DietPi-Set_hardware | Setting in /boot/dietpi.txt adjusted: AUTO_SETUP_HEADLESS=1
[  OK  ] headless enable | Completed
Powering off HDMI
[ INFO ] DietPi-Patch | Patching 6.26 to 6.27
[ INFO ] DietPi-Patch | Removing obsolete DietPi files that could have remained on older systems...
removed '/boot/dietpi/login'
[ INFO ] DietPi-Patch | Patching 6.27 to 6.28
[ INFO ] DietPi-Patch | Patching 6.28 to 6.29
[ INFO ] DietPi-Patch | Re-allowing PHP7.3 when phpBB is installed, supported since v3.3.0
[ INFO ] DietPi-Patch | Translating old DietPi-LED_control settings into new udev rules
[  OK  ] DietPi-Patch | Added setting SUBSYSTEM=="leds", KERNEL=="led0", ACTION=="add", ATTR{trigger}="mmc0" to end of file /etc/udev/rules.d/dietpi-led_control.rules
[  OK  ] DietPi-Patch | Added setting SUBSYSTEM=="leds", KERNEL=="led1", ACTION=="add", ATTR{trigger}="input" to end of file /etc/udev/rules.d/dietpi-led_control.rules
[ INFO ] DietPi-Patch | Removing old DietPi-LED_control config and script location
removed '/boot/dietpi/.dietpi-led_control'
removed '/boot/dietpi/func/dietpi-led_control'
[ INFO ] DietPi-Patch | Installing vmtouch as replacement for DietPi-RAMdisk
[  OK  ] DietPi-Patch | wget https://dietpi.com/downloads/binaries/stretch/vmtouch_armv7l.deb
[ INFO ] DietPi-Patch | APT install for: ./vmtouch_armv7l.deb, please wait...
Selecting previously unselected package vmtouch.
(Reading database ... 28131 files and directories currently installed.)
Preparing to unpack .../vmtouch_armv7l.deb ...
Unpacking vmtouch (1.3.1-dietpi1) ...
Setting up vmtouch (1.3.1-dietpi1) ...
Configuring vmtouch systemd service ...
Created symlink /etc/systemd/system/local-fs.target.wants/vmtouch.service → /lib/systemd/system/vmtouch.service.
[  OK  ] DietPi-Patch | APT install for: ./vmtouch_armv7l.deb
[ INFO ] DietPi-Patch | Removing obsolete flag files
removed '/boot/dietpi/.update_available'
removed '/boot/dietpi/.timesync_exit_status'
removed '/boot/dietpi/.network'
removed '/tmp/.dietpi_motd'
[ INFO ] DietPi-Patch | Migrating WiFi country code to be applied via native crda facility
nl80211 not found.
[  OK  ] DietPi-Patch | Desired setting in /etc/default/crda was already set: REGDOMAIN=
[  OK  ] DietPi-Patch | Setting in /boot/dietpi.txt adjusted: AUTO_SETUP_NET_WIFI_COUNTRY_CODE=
[ INFO ] DietPi-Patch | Removing obsolete dietpi.txt entries
[ INFO ] DietPi-Patch | Removing obsolete dietpi-software config directory
removed '/var/lib/dietpi/dietpi-software/conf/cava.conf'
removed directory '/var/lib/dietpi/dietpi-software/conf'
[ INFO ] DietPi-Patch | Removing obsolete hint about /DietPi RAMdisk from /boot config files
[ INFO ] DietPi-Patch | Patching 6.29 to 6.30
[ INFO ] DietPi-Patch | Patching 6.30 to 6.31
[  OK  ] DietPi-Patch | chmod a+x /mnt/dietpi_userdata
[  OK  ] DietPi-Patch | mkdir /media
vm.swappiness = 1
net.ipv4.ping_group_range = 0 2147483647
[ INFO ] DietPi-Patch | Patching 6.31 to 6.32
removed '/etc/apt/apt.conf.d/99-dietpi-norecommends'
removed '/etc/apt/apt.conf.d/98-dietpi-no_translations'
removed '/etc/apt/apt.conf.d/99-dietpi-forceconf'
[ INFO ] Updating asix ax88179 driver for kernel 4.19.66-v7+, as provided by allo.com:
[ INFO ] - https://github.com/allocom/USBridgeSig/tree/master/ethernet
[ INFO ] Downloading driver...
--2020-10-06 14:54:56--  http://3.230.113.73:9011/Allocom/USBridgeSig/rpi-usbs-4.19.66-v7+/ax88179_178a.ko
Connecting to 3.230.113.73:9011... connected.
HTTP request sent, awaiting response... 200 OK
Length: 44404 (43K)
Saving to: ‘/tmp/ax88179_178a.ko’

     0K .......... .......... .......... .......... ...       100% 94.9K=0.5s

2020-10-06 14:54:57 (94.9 KB/s) - ‘/tmp/ax88179_178a.ko’ saved [44404/44404]

[ INFO ] Installing driver...
removed '/lib/modules/4.19.66-v7+/kernel/drivers/net/usb/ax88179_178a.ko'
'/tmp/ax88179_178a.ko' -> '/lib/modules/4.19.66-v7+/kernel/drivers/net/usb/ax88179_178a.ko'
[ INFO ] Running depmod...
[ INFO ] Cleaning up...
removed '/tmp/ax88179_178a.ko'
[  OK  ] DietPi-Patch | Commenting "arm_freq_min" in config.txt until related system instabilities have been solved
[  OK  ] DietPi-Patch | Removing deprecated "max_usb_current" from config.txt
[  OK  ] DietPi-Software | Initialised database
[  OK  ] DietPi-Software | Reading database
[ SUB4 ] DietPi-Software > Automated reinstall
[ INFO ] DietPi-Software | 115: Webmin is not currently installed
[ INFO ] DietPi-Software | Use "dietpi-software install 115" to install Webmin.
[  OK  ] DietPi-Software | No changes applied for: Webmin
[ INFO ] DietPi-Software | 132: Aria2 is not currently installed
[ INFO ] DietPi-Software | Use "dietpi-software install 132" to install Aria2.
[  OK  ] DietPi-Software | No changes applied for: Aria2
[ INFO ] DietPi-Software | 50: Syncthing is not currently installed
[ INFO ] DietPi-Software | Use "dietpi-software install 50" to install Syncthing.
[  OK  ] DietPi-Software | No changes applied for: Syncthing
[ INFO ] DietPi-Software | 77: Grafana is not currently installed
[ INFO ] DietPi-Software | Use "dietpi-software install 77" to install Grafana.
[  OK  ] DietPi-Software | No changes applied for: Grafana
[ INFO ] DietPi-Software | 90: phpMyAdmin is not currently installed
[ INFO ] DietPi-Software | Use "dietpi-software install 90" to install phpMyAdmin.
[  OK  ] DietPi-Software | No changes applied for: phpMyAdmin
[ INFO ] DietPi-Software | 147: Jackett is not currently installed
[ INFO ] DietPi-Software | Use "dietpi-software install 147" to install Jackett.
[  OK  ] DietPi-Software | No changes applied for: Jackett
[ INFO ] DietPi-Software | 163: GMediaRender is not currently installed
[ INFO ] DietPi-Software | Use "dietpi-software install 163" to install GMediaRender.
[  OK  ] DietPi-Software | No changes applied for: GMediaRender
[ INFO ] DietPi-Software | 98: HAProxy is not currently installed
[ INFO ] DietPi-Software | Use "dietpi-software install 98" to install HAProxy.
[  OK  ] DietPi-Software | No changes applied for: HAProxy
[ INFO ] DietPi-Software | 108: Amiberry is not currently installed
[ INFO ] DietPi-Software | Use "dietpi-software install 108" to install Amiberry.
[  OK  ] DietPi-Software | No changes applied for: Amiberry
[ INFO ] DietPi-Software | 160: Allo_update is not currently installed
[ INFO ] DietPi-Software | Use "dietpi-software install 160" to install Allo_update.
[  OK  ] DietPi-Software | No changes applied for: Allo_update
[ INFO ] DietPi-Patch | APT autopurge, please wait...
(Reading database ... 28136 files and directories currently installed.)
Removing libhavege1:armhf (1.9.1-5+deb9u1) ...
Processing triggers for libc-bin (2.24-11+deb9u4) ...
[  OK  ] DietPi-Patch | APT autopurge
[  OK  ] DietPi-Update | Incremental patching to v6.32.2 completed
```
</details>

**Comment by MichaIng** [comment]
Okay, great, everything is fine then. Luckily there are not much systems on v6.25 anymore, since a reliable fix on a 6.25 system prior to the update is not reliably possible. We could hack something inside via MOTD but not everyone has it enabled.
